frame/contracts/src/tests.rs

```diff
@@ -339,19 +339,19 @@ impl Convert<Weight, BalanceOf<Self>> for Test {
 /// A filter whose filter function can be swapped at runtime.
 pub struct TestFilter;
 
-thread_local! {
-	static CALL_FILTER: RefCell<fn(&Call) -> bool> = RefCell::new(|_| true);
+parameter_types! {
+	static CallFilterTestValue: fn(&Call) -> bool = (|_| true);
 }
 
 impl TestFilter {
 	pub fn set_filter(filter: fn(&Call) -> bool) {
-		CALL_FILTER.with(|fltr| *fltr.borrow_mut() = filter);
+		CallFilterTestValue::mutate(|fltr| *fltr = filter);
 	}
 }
 
 impl Contains<Call> for TestFilter {
 	fn contains(call: &Call) -> bool {
-		CALL_FILTER.with(|fltr| fltr.borrow()(call))
+		CallFilterTestValue::get()(call)
```

Comment:
I don't see why you postfixed everything with `TestValue`? I suggested it _only_ when there is a naming conflict 😅

Reply:
Wasn't everything, they had conflicts